sleap/gui/suggestions.py

```diff
@@ -291,6 +292,57 @@ def _velocity_video(
 
         return cls.idx_list_to_frame_list(frame_idxs, video)
 
+    def max_point_displacement(
+        cls,
+        labels: "Labels",
+        videos: List[Video],
+        displacement_threshold: float,
+        **kwargs,
+    ):
+        """Finds frames with maximum point displacement above a threshold."""
+
+        proposed_suggestions = []
+        for video in videos:
+            proposed_suggestions.extend(
+                cls._max_point_displacement_video(video, labels, displacement_threshold)
+            )
+
+        suggestions = VideoFrameSuggestions.filter_unique_suggestions(
+            labels, videos, proposed_suggestions
+        )
+
+        return suggestions
+
+    @classmethod
+    def _max_point_displacement_video(
+        cls, video: Video, labels: "Labels", displacement_threshold: float
+    ):
+        lfs = labels.find(video)
+        frames = len(lfs)
+
+        if frames < 2:
+            return []
+
+        displacements = []
+        for i in range(1, frames):
+            prev_lf = lfs[i - 1]
+            curr_lf = lfs[i]
+            prev_points = np.array([inst.points_array for inst in prev_lf.instances_to_show])
+            curr_points = np.array([inst.points_array for inst in curr_lf.instances_to_show])
+
+            if prev_points.shape != curr_points.shape:
+                continue
+
+            displacement = np.linalg.norm(curr_points - prev_points, axis=2).sum()
```

Comment:
This calculation for displacement will not gaurantee that we compare the instances we want to because `LabeledFrame.instances_to_show` returns `Instance`s in the order that they were added to the `LabeledFrame` - i.e. there is no gaurantee that `prev_lf.instances_to_show[0]` is annotating the same animal as `curr_lf.instances_to_show[0]`.

Instead, we should be finding `Instance`s that are assigned to the same `Track` - which will guarantee that they are annotating the same animal (or at least attempting to annotate the same animal).

The `Labels.numpy` method returns us a numpy array for all `LabeledFrame`s and all `Instance`s sorted by `Track`:
https://github.com/talmolab/sleap/blob/14c21b4459773c99b941b77867d5746becc68ff7/sleap/io/dataset.py#L2442-L2468